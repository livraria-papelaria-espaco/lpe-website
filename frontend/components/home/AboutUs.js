import { Container, Paper, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';
import PropTypes from 'prop-types';
import BackgroundImg from '~/assets/background.jpg';
import Markdown from '~/components/text/Markdown';

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'relative',
    background: `url(${BackgroundImg}) center right no-repeat fixed padding-box content-box`,
    color: theme.palette.primary.contrastText,
    textAlign: 'center',
  },
  container: {
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
  },
  paper: {
    textAlign: 'justify',
    padding: theme.spacing(2),
  },
}));

const AboutUs = ({ text }) => {
  const classes = useStyles();

  return (
    <div className={classes.root} id='about-us'>
      <Container fixed className={classes.container}>
        <Typography variant='h2' gutterBottom>
          Sobre n√≥s
        </Typography>
        <Paper className={classes.paper}>
          <Markdown>{text}</Markdown>
        </Paper>
      </Container>
    </div>
  );
};

AboutUs.propTypes = {
  text: PropTypes.string,
};

AboutUs.defaultProps = {
  text: '',
};

export default AboutUs;
