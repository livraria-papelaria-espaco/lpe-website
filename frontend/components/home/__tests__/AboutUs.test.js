import React from 'react';
import { shallow } from 'enzyme';
import AboutUs from '../AboutUs';

describe('AboutUs (components/home/AboutUs)', () => {
  it('Should render correctly and match snapshot', () => {
    const text = 'Lorem Ipsum Text';

    const wrapper = shallow(<AboutUs text={text} />);

    expect(wrapper).toMatchSnapshot();
  });
});
