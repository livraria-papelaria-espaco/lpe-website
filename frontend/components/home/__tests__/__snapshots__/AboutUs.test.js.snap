// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs (components/home/AboutUs) Should render correctly and match snapshot 1`] = `
<div
  className="makeStyles-root-1"
  id="about-us"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
    fixed={true}
  >
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="h2"
    >
      Sobre n√≥s
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-3"
    >
      <Markdown
        className=""
      >
        Lorem Ipsum Text
      </Markdown>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
